pkg/apis/controlplane/types.go

```diff
@@ -38,6 +38,8 @@ type PodReference struct {
 	Name string
 	// The namespace of this pod.
 	Namespace string
+	// The UID of this pod.
+	UID types.UID
```

Comment:
Is it for looking up a Pod/ExternalEntity's Group association by Pod/Entity UID? If not necessary, I would avoid adding another (redundant field) for messages between controller and agents.
Probably I should first ask why we need to support look up by UID but not by Namespace + name?

Reply:
I believe it was to make the query similar to group membership, where UID will be used to identify the Group to be queried. In addition @antoninbas and I think it would make sense to move "endpoint" query to use K8s API, similar to other controlplane resources. In that case, we would also move from current "?name=&namespace=" queries to using UID as query parameters.

Reply:
But if you think the UID field is too much an overhead, we can always revert the last commit in this PR and use an additional indexer to solve the problem.

Reply:
But in K8s, the convention is to use Name to identify a resource, which is also easier for end users. Any reason we must use UID?

Reply:
There is no reason we must use UID. If we wanted to go with namespace/name for `groupassociation` query, then I think we probably also need to add back the `groupmembership` endpoint so that users can query group members by clustergroup name instead of namespace as well. 
But I am not sure how that would work once we add namespaced groups. To be precise, `(r *REST) NamespaceScoped()` for `groupmembership` will be hard to define, and I think having two separate endpoints for ClusterGroup and namespaced Groups is not preferable. 
@jianjuns  so do you think using namespace/name for pod/EE groupassociation query while using uid for groupmembership query is okay? @abhiraut and @antoninbas as well.

Reply:
Agree with @jianjuns namespace and name should be used in API to keep align with K8s API convention.
Not sure if I understand the issue you mentioned for groupmembership correctly. My thinking about the groupassociation and groupmembership is:
1. groupassociation should use the pod as the subject. e.g. /apis/v1/groupassociation/namespace/default/podA to query pod "default/podA"'s group association, just like rolebindings normally use role as the subject.
2. groupmembership should use the group as the subject. e.g. apis/v1/clustergroupmembership/groupA to query clustergroup "groupA"'s members, and if there's namespace scoped group in future, it should be another API like apis/v1/groupmembership/namespace/default/groupB.

Use separate APIs for cluster scope and namespace scope resources is common in K8s, for instances clusterroles, clusterrolebindings, roles, rolebindings, and Antrea also has ClusterNetworkPolicyStats and NetworkPolicyStats.

Reply:
Sure, if everyone is on board with this approach, I will modify the PR accordingly.

Reply:
The proposed approach looks good to me.